externalpolicy/counter_party_properties_test.go

```diff
@@ -686,6 +686,86 @@ func Test_merge4(t *testing.T) {
+	}
+
+	ce = new(ConstraintExpression)
+	(*ce) = append((*ce), "prop == value && prop2 == value2")
+	if rp, err := RequiredPropertyFromConstraint(ce); err != nil {
+		t.Errorf("Error: unable to convert simple expression: %v", err)
+	} else if tle := rp.TopLevelElements(); tle == nil {
+		t.Errorf("Error: There should be a top level array element")
+	} else if len(tle) != 1 {
+		t.Errorf("Error: Should be 1 top level array alement")
+	}
+
+	ce = new(ConstraintExpression)
+	(*ce) = append((*ce), "prop == value && prop2 == value2 || prop3 == value3")
+	if rp, err := RequiredPropertyFromConstraint(ce); err != nil {
+		t.Errorf("Error: unable to convert simple expression: %v", err)
+	} else if tle := rp.TopLevelElements(); tle == nil {
+		t.Errorf("Error: There should be 2 top level array elements")
+	} else if len(tle) != 1 {
+		t.Errorf("Error: Should be 1 top level array alement")
+	}
+
+	ce = new(ConstraintExpression)
+	(*ce) = append((*ce), "prop == value && prop2 == value2 || prop3 == value3 || prop4 == value4")
+	if rp, err := RequiredPropertyFromConstraint(ce); err != nil {
+		t.Errorf("Error: unable to convert simple expression: %v", err)
+	} else if tle := rp.TopLevelElements(); tle == nil {
+		t.Errorf("Error: There should be 3 top level array elements")
+	} else if len(tle) != 1 {
+		t.Errorf("Error: Should be 1 top level array alement len(tle): %v tle: %v", len(tle), tle)
+	}
+
+	ce = new(ConstraintExpression)
+	(*ce) = append((*ce), "prop == value && prop2 == value2 || prop3 == value3 || prop4 == value4 && prop5 == value5")
+	if rp, err := RequiredPropertyFromConstraint(ce); err != nil {
+		t.Errorf("Error: unable to convert simple expression: %v", err)
+	} else if tle := rp.TopLevelElements(); tle == nil {
+		t.Errorf("Error: There should be 3 top level array elements")
+	} else if len(tle) != 1 {
+		t.Errorf("Error: Should be 1 top level array alement")
+	}
+}
+
+func Test_complex_conversion(t *testing.T) {
+	ce := new(ConstraintExpression)
+	(*ce) = append((*ce), "prop == value && prop2 == value2", "prop3 == value3 || prop4 == value4 && prop5 == value5")
+	if rp, err := RequiredPropertyFromConstraint(ce); err != nil {
+		t.Errorf("Error: unable to convert simple expression: %v", err)
+	} else if tle := rp.TopLevelElements(); tle == nil {
+		t.Errorf("Error: There should be 3 top level array elements")
+	} else if len(tle) != 2 {
+		t.Errorf("Error: Should be 1 top level array element")
+	}
+
+	ce = new(ConstraintExpression)
+	(*ce) = append((*ce), "prop == value && prop2 == value2", "prop3 == value3 || prop4 == value4 && prop5 == value5", "prop6 >= val6 || prop7 < val7")
+	if rp, err := RequiredPropertyFromConstraint(ce); err != nil {
+		t.Errorf("Error: unable to convert simple expression: %v", err)
+	} else if tle := rp.TopLevelElements(); tle == nil {
+		t.Errorf("Error: There should be 3 top level array elements")
+	} else if len(tle) != 3 {
+		t.Errorf("Error: Should be 1 top level array element")
+	}
+}
+
```

Comment:
Can you use the following as a test case? It is from test_language_test.go 
````
    constraintStrings := []string{
        "iame2edev == true && cpu == 3 || memory <= 32",
        "hello == \"world\"",
        "hello in \"'hi world', 'test'\"",
        "eggs == \"truck load\" AND certification in \"USDA, Organic\"",
        "version == 1.1.1 OR USDA == true",
        "version in [1.1.1,INFINITY) OR cert == USDA",
    }
````
You need to add support for new type version and string array and also string 'in' operator in counterparty_propteties.go. And please use IsSatisfiedBy() function from constraint_expression.go to validate the conversion.